File: frontend/src/types/product.ts
export interface Category {
  id: number;
  name: string;
  icon?: string;
  image?: string;
  parent_id?: number;
  sort_order: number;
  is_active: boolean;
  children?: Category[];
  created_at: string;
  updated_at: string;
}

export interface SKU {
  id: number;
  product_id: number;
  name: string;
  spec: Record<string, string>;
  price: number;
  stock: number;
  image?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Product {
  id: number;
  name: string;
  description: string;
  detail?: string;
  category_id: number;
  category?: Category;
  images: string[];
  cover_image: string;
  base_price: number;
  is_featured: boolean;
  is_hot: boolean;
  is_active: boolean;
  sales_count: number;
  sort_order: number;
  skus: SKU[];
  min_price?: number;
  max_price?: number;
  created_at: string;
  updated_at: string;
}

export interface Banner {
  id: number;
  title: string;
  image: string;
  link?: string;
  sort_order: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ProductQuery {
  page?: number;
  page_size?: number;
  category_id?: number;
  keyword?: string;
  sort?: 'default' | 'price_asc' | 'price_desc' | 'sales';
  is_featured?: boolean;
  is_hot?: boolean;
  min_price?: number;
  max_price?: number;
}

export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: backend/app/schemas/product.py
"""商品相关 Pydantic Schema"""
from typing import Optional
from datetime import datetime

from pydantic import BaseModel, Field


class SKUResponse(BaseModel):
    id: int
    product_id: int
    sku_code: str
    sku_name: str
    specs: dict
    price: float
    stock: int
    image_url: Optional[str] = None

    class Config:
        from_attributes = True


class CategoryResponse(BaseModel):
    id: int
    name: str
    slug: str
    icon_url: Optional[str] = None
    parent_id: Optional[int] = None
    sort_order: int = 0
    is_active: bool = True
    children: list["CategoryResponse"] = []

    class Config:
        from_attributes = True


class BannerResponse(BaseModel):
    id: int
    title: Optional[str] = None
    image_url: str
    link_url: Optional[str] = None
    sort_order: int = 0
    is_active: bool = True

    class Config:
        from_attributes = True


class ProductResponse(BaseModel):
    id: int
    name: str
    slug: str
    description: Optional[str] = None
    category_id: int
    brand: str = "oPhone"
    is_active: bool = True
    is_featured: bool = False
    sales_count: int = 0
    min_price: Optional[float] = None
    cover_image: Optional[str] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class ProductDetail(ProductResponse):
    category: Optional[CategoryResponse] = None
    skus: list[SKUResponse] = []
    images: list["ProductImageResponse"] = []

    class Config:
        from_attributes = True


class ProductImageResponse(BaseModel):
    id: int
    product_id: int
    image_url: str
    sort_order: int = 0

    class Config:
        from_attributes = True


class ProductListResponse(BaseModel):
    items: list[ProductResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
